rllib/algorithms/algorithm.py

```diff
@@ -1735,13 +1735,19 @@ def import_policy_model_from_h5(
         self._sync_weights_to_workers(worker_set=self.workers)
 
     @override(Trainable)
-    def save_checkpoint(self, checkpoint_dir: str) -> str:
-        checkpoint_path = os.path.join(
-            checkpoint_dir, "checkpoint-{}".format(self.iteration)
-        )
-        pickle.dump(self.__getstate__(), open(checkpoint_path, "wb"))
+    def save_checkpoint(self, checkpoint_dir: str) -> Optional[Union[str, Dict]]:
```

Comment:
New AIR way of doing things:
- override tune.Trainable.save_checkpoint
- return state dict (instead of writing to file ourselves). Tune will handle the rest.

Reply:
simply returning self.__getstate__() is great.

a high level question is that why do we want to return the native model files as part of this dict??
why don't we simply write the model files in checkpoint_dir here and call it a day?
I think directly writing files in the dir is still a supported mode right?

the benefit of doing things this way is that:
1. you pretty much don't have to touch any other file/code for this additional export feature.
2. we can avoid having dir_contents_to_dict / dict_to_dir utils. I am having very mixed feelings about this dir to dict thing ...

Reply:
Ah, sorry, this is outdated:

We DO now write everything to disk and DO NOT return a state dict, but the path to the checkpoint dir (which has then all the files already written to it).